src/frontend/components/administration/WeekHistory.vue

```diff
@@ -0,0 +1,111 @@
+<template>
+  <v-card class="my-3">
+    <v-toolbar color="secondary" dark flat dense>
+      <v-toolbar-title v-text="$t('reportHistory.title')" />
+      <v-spacer />
+      <v-col class="text-right">
+        <v-btn
+          @click.stop="setDialogVisible(true)"
+          v-text="$t('update')"
+        />
+      </v-col>
+      <UpdateDialog :dialog="dialogVisible" @closed="setDialogVisible(false)" />
+    </v-toolbar>
+    <v-row v-if="reports.length === 0" align="center" justify="center">
+      <v-col class="text-center" cols="12" sm="4">
+        {{ $t("reportHistory.noReport") }}
+      </v-col>
+    </v-row>
+    <v-row v-else>
+      <v-col
+        v-for="report in reports"
+        :id="report.id"
+        :key="report.id"
+        cols="12"
+        sm="12"
+        md="6"
+        lg="3"
+      >
+        <ReportCard :report="report" :status="getStatusOfReport(report)" />
```

Comment:
Methods are not supposed to be used to populate props. Please use computed properties in the `ReportCard` component instead